privatek8s.tf

```
@@ -169,6 +182,55 @@ resource "azurerm_kubernetes_cluster_node_pool" "infraciarm64" {
   tags = local.default_tags
 }
 
+# nodepool dedicated for the infra.ci.jenkins.io controller
+resource "azurerm_kubernetes_cluster_node_pool" "infracictrlarm64" {
+  name                  = "arm64infracictrl"
+  vm_size               = "Standard_D4pds_v5" # 4 vCPU, 16 GB RAM, local disk: 150 GB and 19000 IOPS
+  os_disk_type          = "Ephemeral"
+  os_disk_size_gb       = 150 # Ref. Cache storage size at https://learn.microsoft.com/en-us/azure/virtual-machines/dpsv5-dpdsv5-series#dpdsv5-series (depends on the instance size)
+  orchestrator_version  = local.kubernetes_versions["privatek8s"]
+  kubernetes_cluster_id = azurerm_kubernetes_cluster.privatek8s.id
+  enable_auto_scaling   = true
+  min_count             = 1
+  max_count             = 3
+  zones                 = [1] # zone 1 is dedicated to ARM64 by azure
+  vnet_subnet_id        = data.azurerm_subnet.privatek8s_infra_ci_controller_tier.id
+
+  node_taints = [
+    "jenkins=infra.ci.jenkins.io:NoSchedule",
+    "kind=infra.ci.jenkins.io"
+  ]
+  lifecycle {
+    ignore_changes = [node_count]
+  }
+
+  tags = local.default_tags
+}
+
+# nodepool dedicated for the release.ci.jenkins.io controller
+resource "azurerm_kubernetes_cluster_node_pool" "releasecictrlarm64" {
+  name                  = "arm64releasecictrl"
```

As per https://learn.microsoft.com/en-us/azure/aks/create-node-pools#limitations, the Linux node pool names are limited to 12 chars max. As @smerle33 reminded me, the Azure API does not validate the name during the "terraform plan" but only at the "terraform apply".

The `arm64releasecictrl` is 18 characters which will fail once the PR is merged.

As such, I propose the following with 11 characters (room to add a suffix number if needed):

```suggestion
resource "azurerm_kubernetes_cluster_node_pool" "releaseci_controller" {
  name                  = "releacictrl"
```

- Terraform resource identifier has no character limitation such as Azure API: we can use readable names
- Removing `arm64` from both Terraform Resource identifier and the node pool name: we only indicate the node pool role (e.g. "hosting infra.ci controller" or "hosting release.ci controller")
- Keeping the `ctrl` (see above)

WDYT?